app/src/main/java/com/lightricks/feedexercise/database/DataAccessObjects.kt

```diff
@@ -1,5 +1,20 @@
 package com.lightricks.feedexercise.database
 
-/***
- * todo: add Room's Data Access Object interface(s) here
- */
+import androidx.lifecycle.LiveData
+import androidx.room.Dao
+import androidx.room.Insert
+import androidx.room.OnConflictStrategy
+import androidx.room.Query
+import io.reactivex.Completable
+
+@Dao
+interface FeedEntitiesDao {
+    @Insert(onConflict = OnConflictStrategy.IGNORE)
```

Comment:
I think it makes more sense to override the record. In the end we want to be able to refresh the feed.

Reply:
Disagree. If we want to add some update functionality, then we should add some timestamp. Overriding when in 99% of the time it's just the same record - makes the database redundant. If we always invalidate most of the records, it's cheaper to just save a zipped raw JSON on disk than to use a dBase.

Reply:
The feed in a real app will be much larger and will change frequently. Some items will change the title or the thumbnail. And parsing the JSON every time is expensive.